src/openclon12/resource_tasks.cpp

```diff
@@ -2532,8 +2532,18 @@ class MapCopyTask : public MapTask
         Args.m_appDesc.m_Height = args.Height;
         Args.m_appDesc.m_usage = D3D12TranslationLayer::RESOURCE_USAGE_STAGING;
         Args.m_appDesc.m_bindFlags = D3D12TranslationLayer::RESOURCE_BIND_NONE;
-        Args.m_appDesc.m_cpuAcess = D3D12TranslationLayer::RESOURCE_CPU_ACCESS_READ | D3D12TranslationLayer::RESOURCE_CPU_ACCESS_WRITE;
-        Args.m_heapDesc = CD3DX12_HEAP_DESC(0, D3D12_HEAP_TYPE_READBACK);
+
+        if (flags | CL_MAP_WRITE)

```

Comment:
This condition is always true (`<value> | nonzero` is always nonzero). This probably should be `if (flags == CL_MAP_WRITE)`, right? Since read or read-write should use a `READBACK` heap.

Reply:
ACK😂. Update it to flags & CL_MAP_WRITE